src/components/QuestionUpsertAccordion.tsx

```diff
@@ -0,0 +1,48 @@
+import {
+	Accordion,
+	AccordionDetails,
+	AccordionSummary,
+	Stack,
+	Typography
+} from '@mui/material';
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
+import { TextFieldElement } from 'react-hook-form-mui';
+import type { Control } from 'react-hook-form';
+
+import type { RiddleUpsertDetail } from '../utils/Types.ts';
+
+import { FileUploader } from './forms/FileUploader.tsx';
+import { HintsUpsert } from './HintsUpsert.tsx';
+import { AnswersUpsert } from './AnswersUpsert.tsx';
+
+type Props = {
+	index: number;
+	control: Control<RiddleUpsertDetail>;
+};
+
+export const QuestionUpsertAccordion = ({ index, control }: Props) => (
+	<Accordion sx={{ flexGrow: 1 }}>
+		<AccordionSummary expandIcon={<ExpandMoreIcon />}>
+			<Typography variant="h6">Question {index + 1}</Typography>
+		</AccordionSummary>
+		<AccordionDetails>
+			<Stack gap={2} sx={{ minWidth: { md: 400 } }}>
+				<TextFieldElement
+					label="Question text"
+					multiline
+					name={`questions.${index}.questionText`}
+					rows={5}
+					// required
```

Comment:
All required fields should be re-enabled (and this property added to some other components as well) at the end but it's much better like this for testing.